platform-sdk/swirlds-logging/src/main/java/com/swirlds/logging/api/internal/level/HandlerLoggingLevelConfig.java

```diff
@@ -0,0 +1,224 @@
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * A configuration that can be used to configure the logging levels of loggers. This class supports to define levels for
+ * packages or classes. The configuration is read from the {@link Configuration} object.
+ * <p>
+ * Like for example in spring boot a configuration like this can be used:
+ * <p>
+ * {@code logging.level.com.swirlds=DEBUG}
+ * <p>
+ * In that case all packages and classes under the package {@code com.swirlds} will be configured to use the level
+ * DEBUG. If a more specific configuration is defined, that configuration will be used instead. For example:
+ * <p>
+ * {@code logging.level.com.swirlds.logging.api=INFO}
+ * <p>
+ * In that case all packages and classes under the package {@code com.swirlds.logging.api} will be configured to use the
+ * level INFO.
+ */
+public class HandlerLoggingLevelConfig {
+
+    public static final String PROPERTY_LOGGING_LEVEL = "logging.level";
+    public static final String PROPERTY_LOGGING_HANDLER_LEVEL = "logging.handler.%s.level";
+    public static final String PROPERTY_PACKAGE_LEVEL = "%s.";
+    public static final String PROPERTY_LOGGING_MARKER = "logging.marker";
+    public static final String PROPERTY_LOGGING_HANDLER_MARKER = "logging.handler.%s.marker";
+
+    /**
+     * The emergency logger.
+     */
+    private static final EmergencyLogger EMERGENCY_LOGGER = EmergencyLoggerProvider.getEmergencyLogger();
+
+    /**
+     * The cache for the levels.
+     */
+    private final Map<String, ConfigLevel> levelCache;
+
+    /**
+     * The cache for the markers.
+     */
+    private final Map<String, MarkerDecision> markerCache;
+
+    /**
+     * The configuration properties.
+     */
+    private final Map<String, ConfigLevel> levelConfigProperties;
+
+    /**
+     * The prefix for the configuration.
+     */
+    private final String name;
+
+    /**
+     * Creates a new logging level configuration based on the given configuration and handler name.
+     *
+     * @param configuration The configuration.
+     * @param name        The name.
+     */
+    public HandlerLoggingLevelConfig(@NonNull Configuration configuration, @NonNull String name) {
+        this.name = Objects.requireNonNull(name, "name must not be null");
+        this.levelCache = new ConcurrentHashMap<>();
+        this.markerCache = new ConcurrentHashMap<>();
+        this.levelConfigProperties = new ConcurrentHashMap<>();
+        update(configuration);
+    }
+
+    /**
+     * Updates the configuration based on the given configuration. That method can be used to change the logging level
+     * dynamically at runtime.
+     *
+     * @param configuration The configuration.
+     */
+    public void update(final @NonNull Configuration configuration) {
+        Objects.requireNonNull(configuration, "configuration must not be null");
```

Comment:
We must check if the exception is handled in the logging. Logging must never throw an exception